Cannot find module '@vitest/utils/diff' or its corresponding type declarations.

### Describe the bug

After updating vitest to 0.30.0, running `tsc --noEmit` will raise the following errors:

```
node_modules/@vitest/runner/dist/utils.d.ts:5:29 - error TS2307: Cannot find module '@vitest/utils/diff' or its corresponding type declarations.

5 import { DiffOptions } from '@vitest/utils/diff';
                              ~~~~~~~~~~~~~~~~~~~~

node_modules/vitest/dist/index.d.ts:6:37 - error TS2307: Cannot find module '@vitest/snapshot/environment' or its corresponding type declarations.

6 export { SnapshotEnvironment } from '@vitest/snapshot/environment';
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```


### Reproduction


https://codesandbox.io/p/sandbox/practical-darwin-sq0t0x?file=%2Fpackage.json

Run `yarn test` in this repo and you will see the error. 

<img width="1470" alt="image" src="https://user-images.githubusercontent.com/24715727/230839685-530ace0f-1760-4f00-b43a-4ed8d8ae5f38.png">


### System Info

```shell
N/A (Codesandbox environment)
```


### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/snapshot/environment.d.ts
+++ b/packages/snapshot/environment.d.ts
@@ -1,0 +1,1 @@
+export * from './dist/environment.js'

--- a/packages/snapshot/manager.d.ts
+++ b/packages/snapshot/manager.d.ts
@@ -1,0 +1,1 @@
+export * from './dist/manager.js'

--- a/packages/utils/package.json
+++ b/packages/utils/package.json
@@ -29,7 +29,8 @@
   "module": "./dist/index.js",
   "types": "./dist/index.d.ts",
   "files": [
-    "dist"
+    "dist",
+    "*.d.ts"
   ],
   "scripts": {
     "build": "rimraf dist && rollup -c",

--- a/packages/utils/src/descriptors.ts
+++ b/packages/utils/src/descriptors.ts
@@ -1,5 +1,10 @@
-import concordance, { type DisplayOptions } from 'concordance'
+import concordance from 'concordance'
 import { getColors } from './colors'
+
+interface DisplayOptions {
+  theme?: any
+  maxDepth?: number
+}
 
 export function getConcordanceTheme() {
   const c = getColors()
@@ -86,8 +91,4 @@ export function getConcordanceTheme() {
 
 export function diffDescriptors(actual: unknown, expected: unknown, options: DisplayOptions) {
   return concordance.diff(expected, actual, options)
-}
-
-export function formatDescriptor(value: unknown, options: DisplayOptions) {
-  return concordance.formatDescriptor(value, options)
 }

--- a/packages/utils/src/diff.ts
+++ b/packages/utils/src/diff.ts
@@ -1,8 +1,6 @@
 import { getColors } from './colors'
 import { diffDescriptors, getConcordanceTheme } from './descriptors'
 
-export * from './descriptors'
-
 export interface DiffOptions {
   showLegend?: boolean
 }

--- a/packages/utils/src/external.d.ts
+++ b/packages/utils/src/external.d.ts
@@ -1,9 +1,3 @@
 declare module 'concordance' {
-  interface DisplayOptions {
-    theme?: any
-    maxDepth?: number
-  }
-
-  export function diff(expected: unknown, actual: unknown, options?: DisplayOptions): string
-  export function formatDescriptor(descriptor: unknown, options?: DisplayOptions): string
+  export function diff(expected: unknown, actual: unknown, options?: any): string
 }

